libs/utils/automated-aria.js

```diff
@@ -0,0 +1,263 @@
+import { getConfig, getFederatedContentRoot } from './utils.js';
+
+export const getTextBeforeHeader = (block) => {
+  if (!block) return '';
+
+  let text = '';
+  let foundHeader = false;
+
+  const traverse = (node) => {
+    if (foundHeader) return;
+
+    if (node.nodeType === Node.ELEMENT_NODE) {
+      if (/^h[1-2]$/i.test(node.tagName)) {
+        foundHeader = true;
+        return;
+      }
+      node.childNodes.forEach(traverse);
+    } else if (node.nodeType === Node.TEXT_NODE) {
+      text += node.nodeValue;
+    }
+  };
+
+  traverse(block);
+  return foundHeader ? text.trim() : '';
+};
+
+export const getProduct = (text, productNames) => {
+  if (!text) return '';
+  const normalizedText = text.toLowerCase();
+  return productNames.find(({ us }) => normalizedText.includes(us.toLowerCase()))?.us || '';
+};
+
+// Helper to extract prefix or suffix from class name parts
+const getPatternPart = (parts, usePrefix) => (usePrefix ? parts[0] : parts[parts.length - 1]);
+
+// Helper to check if two divs share a common class pattern
+const divsSharePattern = (div1, div2, hasCommonPrefix, hasCommonSuffix) => {
+  const div1Classes = Array.from(div1.classList);
+  const div2Classes = Array.from(div2.classList);
+  return div1Classes.some((cn1) => div2Classes.some((cn2) => {
+    const parts1 = cn1.split('-');
+    const parts2 = cn2.split('-');
+    const hasPrefixMatch = hasCommonPrefix && parts1[0] === parts2[0];
+    // eslint-disable-next-line max-len
+    const hasSuffixMatch = hasCommonSuffix && parts1[parts1.length - 1] === parts2[parts2.length - 1];
+    return parts1.length > 1 && parts2.length > 1 && (hasPrefixMatch || hasSuffixMatch);
+  }));
+};
+
+// Helper to check if divs have a common pattern (prefix or suffix)
+const hasCommonPattern = (divs, usePrefix) => {
```

Comment:
nit: use object for bool param
```suggestion
const hasCommonPattern = (divs, { usePrefix = false } = {}) => {
```